src/cascadia/TerminalApp/ActionArgs.h

```diff
@@ -241,6 +242,53 @@ namespace winrt::TerminalApp::implementation
             return *args;
         }
     };
+
+    // Possible SplitState values
+    // TODO:GH#2550/#3475 - move these to a centralized deserializing place
+    static constexpr std::string_view VerticalKey{ "vertical" };
+    static constexpr std::string_view HorizontalKey{ "horizontal" };
+    static TerminalApp::SplitState ParseSplitState(const std::string& stateString)
+    {
+        if (stateString == VerticalKey)
+        {
+            return TerminalApp::SplitState::Vertical;
+        }
+        else if (stateString == HorizontalKey)
+        {
+            return TerminalApp::SplitState::Horizontal;
+        }
+        // default behavior for invalid data
+        return TerminalApp::SplitState::None;
+    };
+
+    struct SplitPaneArgs : public SplitPaneArgsT<SplitPaneArgs>
+    {
+        SplitPaneArgs() = default;
+        GETSET_PROPERTY(winrt::TerminalApp::SplitState, SplitStyle, winrt::TerminalApp::SplitState::None);
+
+        static constexpr std::string_view StyleKey{ "style" };

```

Comment:
nit: style vs direction? does direction imply "split to my left" where style doesn't?

Reply:
See, I thought that too. I also wanted to try and use a different key, since "direction" is already used for "up,down,left,right", while this is a totally different enum. Make it clearer to the user that this should have a different value. 

At least that was my thought.

Granted, how does this mesh with `{"action": "splitPane", "type":"duplicate", "style":"vertical" }` from #998/#3586?
vs:
`{"action": "splitPane", "profile":"{my-guid-or-name}", "style":"vertical" }`

with

`{"action": "splitPane", "profile":"{my-guid-or-name}", "type":"duplicate", "style":"vertical" }` probably just throwing a warning, and doing __nothing?__

Reply:
maybe `splitDirection` and `type`? type=`kind`? hmm.

but yeah maybe a warning and [don't connect the binding] for the last case

Reply:
hmm.. style seems fine. idk.

Reply:
To me, `style` sounds like it would change the appearance (i.e.: color scheme, background, etc...). Probably because of CSS. I'd feel more comfortable with either `split` being in the name of the key (which feels weird but isn't ambiguous).

Or what about `orientation`, `dividerOrientation`?

Reply:
```json
{"action": "splitPane", "style":"vertical" } 
{"action": "splitPane", "type":"duplicate", "style":"vertical" } 
{"action": "splitPane", "profile":"{my-guid-or-name}", "style":"vertical" }`
{"action": "splitPane", "profile":"{my-guid-or-name}", "type":"duplicate", "style":"vertical" }
```

```json
{"action": "splitPane", "split":"vertical" } 
{"action": "splitPane", "type":"duplicate", "split":"vertical" } 
{"action": "splitPane", "profile":"{my-guid-or-name}", "split":"vertical" }`
{"action": "splitPane", "profile":"{my-guid-or-name}", "type":"duplicate", "split":"vertical" }
```

```json
{"action": "splitPane", "orientation":"vertical" }
{"action": "splitPane", "type":"duplicate", "orientation":"vertical" }
{"action": "splitPane", "profile":"{my-guid-or-name}", "orientation":"vertical" }`
{"action": "splitPane", "profile":"{my-guid-or-name}", "type":"duplicate", "orientation":"vertical" }
```

Yea I suppose "split" feels weird in my head, but not on paper. I think that's my new favorite of the three. @DHowett-MSFT thoughts?

Reply:
🚢